### DISCUSSION TITLE
On-demand Databases connections using Adonis V4.

### DISCUSSION THREAD

**Discussion by zLupa** [adonisjs/core#1827] [Help]
Hey! it's possible to have on-demand Databases connections using Adonis V4?

My situation is, i have Databases credentials in a master database, so i need to get that credentials from the Master db and then create a connection and get some information from that db, it's possible using Adonis/Lucid v6.1.3?

I can't set the credentials inside `config/database.js`, because they're specific for every request and there's a lot of credentials.

I created a question in [StackOverflow](https://stackoverflow.com/questions/64529983/initialize-a-mysql-connection-in-adonis-v4) too with bounty, if you want some kind of reward, answer there.

I really need that, even a workaround is welcome.

Thanks for all! (:

**Comment by thetutlage** [comment]
Hello @zLupa 

Before, I look into the code to see if there is possible work around for this. I want to ask this question, is there any specific reason for using v4 over v5?

**Comment by zLupa** [comment]
Thanks for the answer! @thetutlage 

I think no, if Adonis v5 support JavaScript for default and has a upgrade guide, i can upgrade.

**Comment by zLupa** [comment]
Hello! any asnwer? @thetutlage

**Comment by ruanitto** [comment]
I have an application running with v4, and multi databases, solved this with an service to create conections on demand:

 ```javascript
async loadDBConfig(id_employer) {
    const cliente_config = {
      client: 'mysql',
      connection: {
        host: null,
        database: null,
        password: null,
        user: null,
        port: 3306,
        connectTimeout: 90000,
      },
      pool: {
        min: 0,
        max: 100,
        idleTimeoutMillis: 3000,
      },
    }

    const dev = Config.get('app.development')

    const emp = await Employer.query().select('db_conection').where('id', id_employer).first()

    const conn = await this.decryptConnection(emp.db_conection)

    if (conn) {
      cliente_config.client = conn.base
      cliente_config.connection.host = dev ? Config.get('database.client.connection.host') : conn.host
      cliente_config.connection.database = conn.database
      cliente_config.connection.password = dev ? Config.get('database.client.connection.password') : conn.password
      cliente_config.connection.user = dev ? Config.get('database.client.connection.user') : conn.username
      cliente_config.connection.port = dev ? Config.get('database.client.connection.port') : conn.port
      Config.set('database.client_' + String(id_employer), client_config)
    }
  }
```
Not the best way, but work.
